### PR TITLE
Add dedicated build job for all targets

### PR DESCRIPTION
**Pull request by kkohbrok**
This PR refactors the CI a little bit and adds cross-compilation for the following targets:

#### Windows
- i686-pc-windows-msvc
- x86_64-pc-windows-msvc
#### macOS
- x86_64-apple-darwin
- aarch64-apple-darwin
#### Linux
- x86_64-unknown-linux-gnu
- i686-unknown-linux-gnu
- aarch64-unknown-linux-gnu
#### Android
- aarch64-linux-android
#### iOS
- aarch64-apple-ios
- aarch64-apple-ios-sim
#### WASM
- wasm32-unknown-unknown

Thus fixing #267, #471, #464 and #463.

Between #529 and this PR, #189 should also be fixed now.

Future work are #531, #532 and #533.

### CODE REVIEW

**Comment by franziskuskiefer** [comment]
@kkohbrok  #264 doesn't look quite right

**Comment by kkohbrok** [comment]
> @kkohbrok #264 doesn't look quite right

Ah, yes. I think I meant #267.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Build
+
+on:
```
Comment (franziskuskiefer) [inline .github/workflows/build.yml:3]: Can you add a `workflow_dispatch` event to all these to manually trigger them on branches?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          # Compile iOS sim on macOS
+          - os: macos-latest
+            arch: aarch64-apple-ios-sim
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      # Install the required target
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```
Comment (franziskuskiefer) [inline .github/workflows/build.yml:50]: Adding a `profile: minimal` should speed things up and should be enough.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -29,54 +29,12 @@ jobs:
       - uses: actions/checkout@v2
         with:
           ref: ${{ github.event.pull_request.head.sha }}
-      - name: Setup MacOS
-        if: matrix.os == 'macos-latest'
+      - name: Tests debug build
```
Comment (franziskuskiefer) [inline .github/workflows/tests.yml:32]: We should be able to test 32-bit Windows and Linux here as well.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, a couple nits.

I'll then also make all of these required checks before merging.

File: .github/workflows/build.yml (hunk 1 above)
Comment (kkohbrok) [reply .github/workflows/build.yml:3]: What do you mean by "all of these"? All workflows? Or just test and build workflows?

File: .github/workflows/tests.yml (hunk 3 above)
Comment (kkohbrok) [reply .github/workflows/tests.yml:32]: I filed #531, #532 and #533.

File: .github/workflows/build.yml (hunk 1 above)
Comment (franziskuskiefer) [reply .github/workflows/build.yml:3]: Everywhere would be nice. But at least in the two that you touch here.

File: .github/workflows/fuzz.yml [hunk 4]
```diff
@@ -7,6 +7,9 @@ on:
   pull_request:
     branches:
       - main
+  workflow_dispatch:
+    branches:
```
Comment (franziskuskiefer) [inline .github/workflows/fuzz.yml:11]: Why only on main? This makes the trigger a little pointless. The idea is that you can press the button on any branch, i.e. code that doesn't trigger the CI automatically.

**Comment by franziskuskiefer** [comment]
This PR should also update the Readme to mention all the supported platforms now.

File: README.md [hunk 5]
```diff
@@ -18,10 +18,22 @@ A WIP Rust implementation of [Messaging Layer Security](https://github.com/mlswg
...
+- x86_64-unknown-linux-gnu
+- i686-unknown-linux-gnu
+- x86_64-pc-windows-msvc
+- i686-pc-windows-msvc
+- x86_64-apple-darwin
+
+In addition, the CI builds (but doesn't test) the following rust targets.
+
+- aarch64-apple-darwin
+- aarch64-unknown-linux-gnu
```
Comment (franziskuskiefer) [inline README.md:32]: This is tested on drone

**Review by franziskuskiefer — DISMISSED** [review]
:shipit:

**Review by raphaelrobert — APPROVED** [review]

**Comment by kkohbrok** [comment]
Closing in favor of #553 to keep a cleaner commit history.

**Pull request closed without merging**